consumer.go

```diff
@@ -401,10 +401,22 @@ func (child *partitionConsumer) HighWaterMarkOffset() int64 {
 }
 
 func (child *partitionConsumer) responseFeeder() {
+feederLoop:
 	for response := range child.feeder {
-		switch err := child.handleResponse(response); err {
+		switch msgs, err := child.parseResponse(response); err {
 		case nil:
-			break
+			for i, msg := range msgs {
+				select {
+				case child.messages <- msg:
+				case <-time.After(child.conf.Consumer.MaxProcessingTime):
+					child.broker.done <- child
+					for _, msg = range msgs[i:] {
+						child.messages <- msg
+					}
+					child.broker.input <- child
+					continue feederLoop
```

Comment:
So, if the messages never get drained by the consumer (because their consumer goroutine shut down for some reason, we will end up with a stuck goroutine?

I think that's OK in that scenario, just want to understand the code.